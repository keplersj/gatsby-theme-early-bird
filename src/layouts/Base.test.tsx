import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Layout from "./Base";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    file: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    backdrop: {
      childImageSharp: {
        fluid: {
          base64: "",
          aspectRatio: 1.333,
          src: "",
          srcSet: "",
          srcWebp: "",
          srcSetWebp: "",
          sizes: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: "",
        description: "",
        siteUrl: "",
        twitterUsername: "",
        instagramUsername: "",
        linkedinUsername: "",
        githubUsername: ""
      }
    }
  }));
});

describe("Base Layout", (): void => {
  it("renders correctly", (): void => {
    const tree = renderer
      .create(
        <Layout>
          <span>Test</span>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
