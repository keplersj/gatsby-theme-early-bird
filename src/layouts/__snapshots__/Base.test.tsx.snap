// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Layout renders correctly 1`] = `
<head>
  Object {}
  Object {
    "lang": "en",
  }
  <meta
    content="Kepler Sticka-Jones"
    data-react-helmet={true}
    name="author"
  />
  <meta
    content=""
    data-react-helmet={true}
    name="description"
  />
  <meta
    content="#409040"
    data-react-helmet={true}
    name="theme-color"
  />
  <meta
    content=""
    data-react-helmet={true}
    property="og:title"
  />
  <meta
    content="website"
    data-react-helmet={true}
    property="og:type"
  />
  <meta
    content=""
    data-react-helmet={true}
    property="og:url"
  />
  <meta
    content=""
    data-react-helmet={true}
    property="og:image"
  />
  <meta
    content="image/jpeg"
    data-react-helmet={true}
    property="og:image:type"
  />
  <meta
    content="480"
    data-react-helmet={true}
    property="og:image:width"
  />
  <meta
    content="480"
    data-react-helmet={true}
    property="og:image:height"
  />
  <meta
    content=""
    data-react-helmet={true}
    property="og:description"
  />
  <meta
    content="summary"
    data-react-helmet={true}
    name="twitter:card"
  />
  <meta
    content="@"
    data-react-helmet={true}
    name="twitter:site"
  />
  <meta
    content="@"
    data-react-helmet={true}
    name="twitter:creator"
  />
  <meta
    content="E5D802A305BE2DD99EB4484A4B1DD90E"
    data-react-helmet={true}
    name="msvalidate.01"
  />
  <title
    data-react-helmet={true}
  >
    
  </title>
</head>

<main>
  <span>
    Test
  </span>
</main>
`;
